src/classes/RD2_OpportunityService.cls

```diff
@@ -225,37 +226,41 @@ public with sharing class RD2_OpportunityService {
     * @return RD2_OpportunityService This Opportunity Service instance
     */
     public RD2_OpportunityService updateOpportunities(List<npe03__Recurring_Donation__c> rds) {
-        return updateOpportunities(rds, new Map<Id, Boolean>());
+        return updateOpportunities(rds, new Set<Id>());
     }
 
     /***
     * @description Updates related future open installment Opps
     * @param rds Recurring Donations
-    * @param isScheduleChangedById Map of Recurring Donation Ids to Boolean indicating whether
+    * @param rdIdsWhereScheduleChanged Recurring Donation Ids for which a schedule has been changed
     * schedule has changed
     * @return RD2_OpportunityService This Opportunity Service instance
     */
-    public RD2_OpportunityService updateOpportunities(List<npe03__Recurring_Donation__c> rds, Map<Id, Boolean> isScheduleChangedById) {
-        List<Opportunity> oppsToUpdate = new List<Opportunity>();
+    public RD2_OpportunityService updateOpportunities(List<npe03__Recurring_Donation__c> rds, Set<Id> rdIdsWhereScheduleChanged) {
+        Map<Id, Opportunity> updatedOppById = new Map<Id, Opportunity>();
         Map<Id, RD2_ScheduleService.Installment> installmentByOppId = new Map<Id, RD2_ScheduleService.Installment>();
 
         //Synchronize Opps with matching Schedule installments
         for (npe03__Recurring_Donation__c rd : rds) {
             RD2_OpportunityMatcher matcher = new RD2_OpportunityMatcher(currentDate)
                 .includeCurrentOrOpen(rd.npe03__Donations__r);
 
+            Integer numberOfInstallments = matcher.getNumberOfOpportunities();
             matcher.match(scheduleService.getVisualizedInstallments(
-                currentDate, matcher.getNumberOfOpportunities(), rd.RecurringDonationSchedules__r
+                currentDate, numberOfInstallments, rd.RecurringDonationSchedules__r
             ));
 
             // Track unmatched matcher records so we can connect Opps and RDs from RDs with altered schedules
-            Boolean openOpportunityMatched = false;
-            Opportunity earliestUnmatchedOpportunity;
+            Boolean isOpenOppMatched = false;
+            Opportunity earliestUnmatchedOpp;
             RD2_ScheduleService.Installment earliestUnmatchedInstallment;
 
             for (RD2_OpportunityMatcher.Record record : matcher.getRecords()) {
+                isOpenOppMatched = false;
```

Comment:
@npsp-reedestockton , should the `isOpenOppMatched` be reset for every matched record or just once for the RD?
CC @force2b

Reply:
Let's leave it as is since current work is passing w/o this line. It can be reviewed in other related PRs.